ui/src/components/editor/shapeComponents/RoleGroupBuilder.vue

```diff
@@ -319,19 +278,24 @@ function validateEntity() {
 
 function isGroupValid(group: any[]): boolean {
   invalid.value = false;
-  if (group.length == 0 || (group.length == 1 && group[0].key["@id"] == IM.GROUP_NUMBER)) {
+  if (group.length == 0 || (group.length == 1 && group[0]?.key?.["@id"] == IM.GROUP_NUMBER)) {
     specificValidationErrorMessage.value = "Role groups can not be empty.";
     return false;
   }
   for (const pair of group) {
+    if (!pair.key) {
+      pair["key"] = { "@id": "", name: "" };
+    }
+    if (!pair.value && pair.value !== 0) {
+      pair["value"] = { "@id": "", name: "" };
+    }
     if (pair.key["@id"] != IM.GROUP_NUMBER) {
-      if (!pair?.key?.["@id"] || pair.key["@id"] == "") {
+      if (!isObjectHasKeys(pair.key, ["iri"]) && (!pair?.key?.["@id"] || pair.key["@id"] == "")) {
         specificValidationErrorMessage.value = "Missing role property.";
         invalid.value = true;
         return false;
       }
-
-      if (!pair?.value?.["@id"] || pair.value["@id"] == "") {
+      if (!isObjectHasKeys(pair.value, ["iri"]) && (!pair?.value?.["@id"] || "" === pair.value["@id"] || null === pair.value["@id"])) {
```

Comment:
extract this check as a method, hasValidIri. Looks like this logic is used multiple times